man/systemd.resource-control.xml

```diff
@@ -1150,9 +1150,16 @@ DeviceAllow=/dev/loop-control
           <para>Allows deprioritizing or omitting this unit's cgroup as a candidate when
           <command>systemd-oomd</command> needs to act. Requires support for extended attributes (see
           <citerefentry project='man-pages'><refentrytitle>xattr</refentrytitle><manvolnum>7</manvolnum></citerefentry>)
-          in order to use <option>avoid</option> or <option>omit</option>. Additionally,
-          <command>systemd-oomd</command> will ignore these extended attributes if the unit's cgroup is not
-          owned by the root user.</para>
+          in order to use <option>avoid</option> or <option>omit</option>.</para>
+
+          <para><command>systemd-oomd</command> will only respect these extended attributes if
+          the unit's cgroup owner, and the owner of the monitored ancestor cgroup are the
+          same. For example, if <filename>-.slice</filename> is monitored by <command>systemd-oomd</command>,
+          then these extended attributes would be ignored if set on descendants of
+          <filename>/user.slice/user-1000.slice/user@1000.service/</filename>. However, if no ancestors of
+          <filename>/user.slice/user-1000.slice/user@1000.service/</filename> are monitored by
```

Comment:
> > However, if no ancestors of /user.slice/user-1000.slice/user@1000.service/ are monitored by...
> 
> This should be allowed as long as the avoid/omit cgroup is a candidate for a monitored cgroup that has the same cgroup owner. So for example, if we have something like this:
> 
> * -.slice # monitoring for pressure threshold 60%
>   
>   * user.slice
>     
>     * user-1000.slice
>       
>       * [user@1000.service](mailto:user@1000.service) # monitoring for pressure threshold 30%
>         
>         * app.slice
>           
>           * my.service # oomd_omit
> 
> xattrs on `my.service` can be respected if it is a candidate for `user@1000.service`. But if we're calculating the candidates for `-.slice`, then oomd xattrs on `my.service` should be ignored. Basically regardless of whether `user@1000.service` has an ancestor being monitored or not. Does that make sense?

Yes, this is a good clarifying example. It is more clear looking at the memory pressure scenario because `oomd_pressure_above` does initial filtering of the cgroups which means (using the example above) `user@1000.service` could be a target, but not `-.slice`. But, with swap kill every monitored cgroup is an initial candidate and will be sorted in `oomd_kill_by_swap_usage`, which led me to the monitored ancestor logic.

> This last part should be reworded to reflect that (maybe remove the stuff about "no ancestors"). I think this might also simplify some of the code such that the check between UIDs would only need to happen during candidate selection/ordering.

Yes, I think this should simplify the code quite a bit. Thanks for taking a look.